lib/kuality-kfs/global_config.rb

```diff
@@ -354,7 +353,11 @@ def find_cg_accounts_in_cg_responsibility_range(lower_limit, upper_limit)
       raise RuntimeError, "No CG Accounts with CG Account Responsibility ID in range #{lower_limit} to #{upper_limit} found."
     elsif accounts_hash.has_key?('org.kuali.kfs.coa.businessobject.Account')  # multiple accounts found
       accounts_array = accounts_hash['org.kuali.kfs.coa.businessobject.Account']
-      accounts_array.each{ |value|
+      # web service can return an overwhelming amount of data, only take up to a max number
+      # value is used more than once so make one call and reuse locally
+      max_rows = (get_aft_parameter_value(ParameterConstants::DEFAULT_MAX_NUM_DATA_ROWS_TO_USE)).to_i
+      accounts_array.length > max_rows ? number_of_elements = max_rows : number_of_elements = accounts_array.length
```

Comment:
Thanks for the second set of eyes.  Implemented your first suggestion.  changes have been made and are available for you to look at again.